service/storage_service.cc

```diff
@@ -5408,11 +5410,12 @@ void storage_service::init_messaging_service(sharded<service::storage_proxy>& pr
                 curr_cdc_gen_id = ss._topology_state_machine._topology.current_cdc_generation_id;
             }
 
-            std::optional<canonical_mutation> cdc_generation_mutation;
+            std::vector<canonical_mutation> cdc_generation_mutations;
             if (curr_cdc_gen_id) {
```

Comment:
I don't think you want to do this check now.
There may be no current CDC generation, but one generation is in the middle of being sent.
If you triggered a snapshot transfer at that moment, this condition would be `false` and the receiver wouldn't obtain the partially applied generation.

It looks like we can get rid of `curr_cdc_gen_id` variable altogether.

Reply:
Fixed.